### PR TITLE
Fix Problem With Some Onward Content Images Not Loading

### LINKED ISSUE #6274: Onward Content Images Not Loading
Example here: https://mobile.guardianapis.com/uk/rendered-items/culture/2022/oct/23/barbra-streisand-interview-bon-soir-new-album

May be related to #6267.

### PR DESCRIPTION
**Pull request by JamieB-gu**
## Why?

Fixes #6274. Some image URLs in onward content are provided on the `static-secure` subdomain. This needs to be translated to the `static` source when building URLs for the Fastly image resizer.

Tested in CODE.

## Changes

- Created image resizer paths with `static`, `uploads` and `media` only
- Added tests for the `src` function
- Updated image tests to use valid domains

### CODE REVIEW

File: apps-rendering/src/image/srcsets.ts [hunk 1]
```diff
@@ -27,7 +27,26 @@ interface Srcsets {
...
 
-const getSubdomain = (domain: string): string => domain.split('.')[0];
+/**
+ * Get the resizing source for a given CAPI asset
+ *
+ * @param url The asset URL from CAPI
+ * @returns An image resizer source
+ */
+const getSource = (url: URL): string => {
+	const subdomain = url.hostname.split('.')[0];
```
Comment (mxdvl) [inline apps-rendering/src/image/srcsets.ts:37]: If you like that kind of stuff…

```suggestion
	const [subdomain] = url.hostname.split('.');
```

File: apps-rendering/src/image/srcsets.ts [hunk 2]
```diff
@@ -27,7 +27,26 @@ interface Srcsets {
...
 // ----- Functions ----- //
 
-const getSubdomain = (domain: string): string => domain.split('.')[0];
+/**
+ * Get the resizing source for a given CAPI asset
+ *
+ * @param url The asset URL from CAPI
+ * @returns An image resizer source
+ */
+const getSource = (url: URL): string => {
```
Comment (mxdvl) [inline apps-rendering/src/image/srcsets.ts:36]:
```suggestion
const getSource = (url: URL) => {
```

I think if you omit the return type, TypeScript will be able to narrow it further for you!

**Review by georgeblahblah — APPROVED** [review]
Looks good to me! I found the tests using `URL` easy to read.

I wonder if it'll be possible at some point to use un-signed image resizer URLs, too (like https://github.com/guardian/dotcom-rendering/pull/6052)

File: apps-rendering/src/image/srcsets.ts (hunk 2 above)
Comment (JamieB-gu) [reply apps-rendering/src/image/srcsets.ts:36]: You're not allowed to omit function return types in AR 🙂:

https://github.com/guardian/dotcom-rendering/blob/e9b3b1bccd75b2385efd3000dcef533e44c7eee8/apps-rendering/config/.eslintrc.js#L87

Comment (mxdvl) [reply apps-rendering/src/image/srcsets.ts:36]: Ah, nice one, so you would have to be explicit.

```suggestion
const getSource = (url: URL): 'static' | 'uploads' | 'media' => {
```

I may be enclined to used an `as const` object in this instance to let the compiler do its thing.

**Pull request merged**